src/compositor/lipstickcompositorwindow.cpp

```diff
@@ -337,7 +348,6 @@ void LipstickCompositorWindow::handleTouchCancel()
         inputDevice->sendTouchCancelEvent();
         inputDevice->setMouseFocus(0, QPointF());
     }
-    window()->removeEventFilter(this);
```

Comment:
Right, so this line should be kept, else we will fail to release the filter in some cases breaking input to other clients. The window() here should also be guarded, i'm seeing in some cases it is NULL.